### PR TITLE
Add `Output.when`

### PR DESCRIPTION
**Pull request by pawelprazak**
fixes https://github.com/VirtusLab/besom/issues/435

### CODE REVIEW

File: core/src/main/scala/besom/internal/Output.scala [hunk 1]
```diff
@@ -80,6 +80,11 @@ trait OutputFactory:
   )(using BuildFrom[CC[Output[B]], B, To], Context): Output[To] = sequence(coll.map(f).asInstanceOf[CC[Output[B]]])
 
   def fail(t: Throwable)(using Context): Output[Nothing] = Output.fail(t)
+
+  def when[A: Typeable](cond: => Boolean | Output[Boolean] | Output[Option[Boolean]])(
```
Comment (lbialy) [inline core/src/main/scala/besom/internal/Output.scala:84]: `Input.Optional[Boolean]` instead

File: core/src/main/scala/besom/internal/Output.scala [hunk 2]
```diff
@@ -80,6 +80,11 @@ trait OutputFactory:
...
+    a: => A | Output[A] | Output[Option[A]]
```
Comment (lbialy) [inline core/src/main/scala/besom/internal/Output.scala:85]: same

File: core/src/test/scala/besom/internal/OutputTest.scala [hunk 3]
```diff
@@ -138,4 +138,46 @@ class OutputTest extends munit.FunSuite:
...
+    given Context = DummyContext().unsafeRunSync()
+
+    val v: Output[String] = Output({
+      fail("This should not be evaluated")
+      "value"
+    })
+    val result = Output.when(false)(v)
+    assertEquals(result.getData.unsafeRunSync(), OutputData(None))
+  }*/
+
```
Comment (lbialy) [inline core/src/test/scala/besom/internal/OutputTest.scala:182]: uh what?

**Review by lbialy — CHANGES_REQUESTED** [review]
one or two things

File: core/src/main/scala/besom/internal/Output.scala (hunk 1 above)
Comment (pawelprazak) [reply core/src/main/scala/besom/internal/Output.scala:84]: done

File: core/src/test/scala/besom/internal/OutputTest.scala (hunk 3 above)
Comment (pawelprazak) [reply core/src/test/scala/besom/internal/OutputTest.scala:182]: this should not be here, removed, thanks

**Review by lbialy — APPROVED** [review]

**Pull request merged**